bot/seasons/easter/easter_riddle.py

```diff
@@ -0,0 +1,84 @@
+import asyncio
+import logging
+import random
+from json import load
+from pathlib import Path
+
+import discord
+from discord.ext import commands
+
+from bot.constants import Colours
+
+log = logging.getLogger(__name__)
+
+with open(Path('bot', 'resources', 'easter', 'easter_riddle.json'), 'r', encoding="utf8") as f:
+    RIDDLE_QUESTIONS = load(f)
+
+TIMELIMIT = 10
+
+
+class EasterRiddle(commands.Cog):
+    """This cog contains the command for the Easter quiz!"""
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.winners = []
+        self.correct = ""
+
+    @commands.command(aliases=["riddlemethis", "riddleme"])
+    async def riddle(self, ctx):
+        """Gives a random riddle questions, then provides 2 hints at 10 second intervals before revealing the answer"""
+        random_question = random.choice(RIDDLE_QUESTIONS)
+        question = random_question["question"]
+        hints = random_question["riddles"]
+        self.correct = random_question["correct_answer"]
+
+        description = f"You have {TIMELIMIT} seconds before the first hint.\n\n"
+
+        q_embed = discord.Embed(title=question, description=description, colour=Colours.pink)
+
+        await ctx.send(embed=q_embed)
+        await asyncio.sleep(TIMELIMIT)
+
+        h_embed = discord.Embed(
+            title=f"Here's a hint: {hints[0]}!",
+            colour=Colours.pink
+        )
+
+        await ctx.send(embed=h_embed)
+        await asyncio.sleep(TIMELIMIT)
+
+        h_embed = discord.Embed(
+            title=f"Here's a hint: {hints[1]}!",
+            colour=Colours.pink
+        )
+
+        await ctx.send(embed=h_embed)
+        await asyncio.sleep(TIMELIMIT)
+
+        if self.winners:
+            win_list = " ".join(self.winners)
+            content = f"Well done {win_list} for getting it correct!"
+            self.winners = []
+        else:
+            content = "Nobody got it right..."
+
+        a_embed = discord.Embed(
+            title=f"The answer is: {self.correct}!",
+            colour=Colours.pink
+        )
+
+        await ctx.send(content, embed=a_embed)
+
+    @commands.Cog.listener()
+    async def on_message(self, message):
```

Comment:
For sanity's sake, this should only check in the Seasonalbot channel for the correct response rather than every channel Seasonalbot has access to.

We have a channel constant defined for this, as `int`, in `bot.constants.Channels.seasonalbot_chat`

Reply:
That does mean that this won't work in bot-commands and off-topic, which are the channels we usually send people to actually use seasonalbot commands.

Reply:
Apologies for the long wait, but I think I've implemented your requested changes.